pkg/threescale/authorizer/authorizer.go

```diff
@@ -0,0 +1,276 @@
+package authorizer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/3scale/3scale-authorizer/pkg/system/v1/cache"
+	"github.com/3scale/3scale-go-client/threescale"
+	"github.com/3scale/3scale-go-client/threescale/api"
+	"github.com/3scale/3scale-porta-go-client/client"
+)
+
+// Authorizer fetches configuration from 3scale and authorizes requests to 3scale
+type Authorizer interface {
+	GetSystemConfiguration(systemURL string, request SystemRequest) (client.ProxyConfig, error)
+	AuthRep(backendURL string, request BackendRequest) (*BackendResponse, error)
```

Comment:
Could you expose: 
* authorize
* report
* authrep ?

Reply:
Yes, will bring them in when we add the backend cache. Thats defiently the idea